.github/workflows/standard_library_tests_and_examples.yml

```diff
@@ -96,6 +96,14 @@ jobs:
 
           # Ensure `FileCheck` from the pre-installed LLVM 15 package is visible
           echo $(brew --prefix llvm@15)/bin/ >> $GITHUB_PATH
+      
+      - name: Validate documentation strings
+        run: |
+          mojo doc -warn-missing-doc-strings -o /dev/null stdlib/src/
+          if [ $? -ne 0 ]; then
```

Comment:
**Suggestion** Unfortunately, this won't work as you hope since the above command still exits with exit code `0` since it produces _warnings_ (not errors).  You can verify this locally:

```bash
 mojo doc -warn-missing-doc-strings -o /dev/null stdlib/src/
Included from /Users/joe/dev/mojo/stdlib/src/collections/__init__.mojo:1:
/Users/joe/dev/mojo/stdlib/src/collections/dict.mojo:49:7: warning: public symbol 'StringableKeyElement' is missing a doc string
trait StringableKeyElement(KeyElement, Stringable):
      ^
Included from /Users/joe/dev/mojo/stdlib/src/os/__init__.mojo:1:
/Users/joe/dev/mojo/stdlib/src/os/os.mojo:274:4: warning: function takes parameters, but no 'Parameters' in doc string
fn remove[pathlike: os.PathLike](path: pathlike) raises:
   ^
Included from /Users/joe/dev/mojo/stdlib/src/os/__init__.mojo:1:
/Users/joe/dev/mojo/stdlib/src/os/os.mojo:298:4: warning: function takes parameters, but no 'Parameters' in doc string
fn unlink[pathlike: os.PathLike](path: pathlike) raises:
   ^
Included from /Users/joe/dev/mojo/stdlib/src/testing/__init__.mojo:1:
/Users/joe/dev/mojo/stdlib/src/testing/testing.mojo:115:4: warning: function takes parameters, but no 'Parameters' in doc string
fn assert_equal[T: Testable](lhs: T, rhs: T, msg: String = "") raises:
   ^
Included from /Users/joe/dev/mojo/stdlib/src/testing/__init__.mojo:1:
/Users/joe/dev/mojo/stdlib/src/testing/testing.mojo:175:4: warning: function takes parameters, but no 'Parameters' in doc string
fn assert_not_equal[T: Testable](lhs: T, rhs: T, msg: String = "") raises:
```

combined with `echo $?` which reports the exit code of `0`.

To "fix" this problem, you can use the `check-file-is-empty.py` script from https://github.com/modularml/mojo/pull/2118 that got removed (just bring it back for this PR).

Longer term, I think we should file a GitHub issue and have a discussion with @River707 and @modocache on their take on this command.  The current structure of producing warnings for invalid API doc strings **and** exiting with an exit code of `0`, in my opinion, pushes complexity to every user (such as with this `check-file-is-empty.py` script).  I'd rather have _some option_ (even if it's `-error-missing-doc-strings` or name to be bikeshedded) which would give me a non-zero exit code if there are any invalid doc strings.  I think that will be the common use case for users of this command.

Reply:
That is correct, i will look into the file and make changes accordingly by looking at the mentioned file.